dashboard/app/models/script_level.rb

```diff
@@ -647,4 +648,11 @@ def get_level_keys(seed_context, use_existing_level_keys = true)
     end
     my_levels.sort_by(&:id).map(&:key)
   end
+
+  # @param [Array<Hash>] levels_data - Array of hashes each representing a level
+  def update_levels(levels_data)
```

Comment:
is this getting the levels to be updated? The name of the method make me think its updating levels but it looks like it just finding the levels to update. Maybe change name to find_levels_to_update or something

Reply:
it is updating the contents of the `levels` association on this ScriptLevel object. it does not update the contents of the Level objects. does that help ? I'm happy to improve the name but not sure that "find_levels_to_update" is right because we are not just finding -- we are actually updating.

Reply:
Sorry you are right I missed the assignment at the front of the line my bad. Please ignore 🤦